docs/developer-docs/integrations/bitcoin/ckbtc.md

```diff
@@ -2,37 +2,476 @@
+- `update_balance`: Instructs the ckBTC minter to check the balance of a Bitcoin address and mint ckBTC into the account of the owner.
+- `estimate_withdrawal_fee`: Returns a current estimate for the fee to be paid when retrieving a certain BTC amount.
+- `get_deposit_fee`: Returns the fee charged when minting ckBTC.
+- `get_withdrawal_account`: Returns a specific ckBTC account where the owner must transfer ckBTC before being able to retrieve BTC.
+- `retrieve_btc`: Instructs the ckBTC minter to burn a certain ckBTC amount and send the corresponding BTC amount, minus fees, to a provided Bitcoin address.
+- `retrieve_btc_status`: Returns the status of a previous retrieve_btc call.
+- `get_minter_info`: Returns information about the ckBTC minter itself.
+- `get_events`: Returns a set of events at the ckBTC minter.
+The endpoints are discussed in more detail in the following.
+
+### `get_btc_address(owner: opt principal, subaccount: opt blob)`
+The provided principal ID and subaccount are concatenated to form the derivation path for the [ecdsa_public_key](https://internetcomputer.org/docs/current/references/ic-interface-spec#ic-ecdsa_public_key) function, which returns the derived public key. If no principal ID is provided, then the sender’s principal ID is used. If no subaccount is provided, then the default subaccount (all zeros) is used.
+
+This public key is encoded as a pay-to-witness-public-key-hash (P2WPKH) Bitcoin address and returned as a text.
+
+Note that the key derivation is not BIP-32 compliant where 31 bits are used for each derivation level. Instead, a single derivation is performed based on the full principal ID and subaccount. Since the derivation is deterministic, a canister can derive the Bitcoin address for a given principal ID and subaccount itself.
+
+### `track_balance(owner: opt principal, subaccount: opt blob)`
+:::info
+This endpoint is work-in-progress and not available yet.
+:::
+
+The ckBTC minter starts tracking the Bitcoin address derived from the provided principal ID and subaccount using the get_btc_address endpoint. If no principal ID is provided, then the sender’s principal ID is used. If no subaccount is provided, then the default subaccount (all zeros) is used.
+
+The balance of the Bitcoin address is not tracked indefinitely. Tracking is stopped if either at least one new unspent transaction output (UTXO) is discovered or there is no new UTXO within a certain time interval (details about balance tracking are provided below).
+
+If at least one new UTXO is discovered, the same amount of ckBTC tokens minus the KYT fee are minted and made available to the owner.
+
+### `update_balance(owner: opt principal, subaccount: opt blob)`
+Instead of having the ckBTC minter track the balance of a Bitcoin address, the update_balance function can be invoked to instruct the ckBTC minter to check if there are new UTXOs for a particular Bitcoin address.
```

Comment:
```suggestion
Instead of having the ckBTC minter track the balance of a Bitcoin address, the `update_balance` function can be invoked to instruct the ckBTC minter to check if there are new UTXOs for a particular Bitcoin address.
```